Common/include/toolboxes/inlet_interpolation_functions.hpp

```diff
@@ -0,0 +1,85 @@
+/*!
+ * \file inlet_interpolation_functions.hpp
+ * \brief Inlet_interpolation_functions
+ * \author Aman Baig
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation 
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include <string>
+#include <cmath>
+#include <vector>
+#include "../config_structure.hpp"
+#include "../../SU2_CFD/include/CMarkerProfileReaderFVM.hpp"
+
```

Comment:
Thanks for commenting on my changes.

I agree that the Interpolator should ideally work as you are saying but in this case I need to fetch the vertex coordinates for the Interpolator to be able to interpolate over those points, that's why I passed the geometry pointer into the class.

I wanted the class to be 'branched' from the main program so that I could read the inlet file in the Interpolator class and not worry about passing the profileReader object from Solver to the Interpolator class.

This class can still be used in other contexts. I may decide to read the profile in the Solver only (now that I think more about it), but even without it its input arguments are generic.

Reply:
> I agree that the Interpolator should ideally work as you are saying but in this case I need to fetch the vertex coordinates for the Interpolator to be able to interpolate over those points, that's why I passed the geometry pointer into the class.

Somewhere in the solver:
```
...
reader = ProfileReader(args)
switch(interpType) {
    case LINEAR:
        interpolator = LinearInterpolator(reader.data())
    case ...
}
for(vertex : marker) {
    interpValues = interpolator(vertex.GetCoord())
    // set the values somewhere
}
...
```
Why is something like that not possible? The profile data contains the coordinates right? Does each interpolated value depend on all "output" locations?

Reply:
This can work. For such an approach, is it possible to just have a set of functions in .cpp and .hpp files which interpolate somewhat like this: LinearInterpolator(reader.data( ) , vertex.GetCoord( )). I have tried this approach in previous commits but abandoned it as I felt that since this subroutine (LoadInletProfile(args)) was changed in the newer v7.0 version, its better to branch out completely with a class rather than accommodate any more changes that happen without any prior notice. Is it ok to have a set of functions in the toolboxes rather than a class?

Reply:
I suppose that would be ok for the implementation methods you have, but it is not ideal because:
- You cannot force a common interface for all methods;
- For some interpolation methods (splines, radial basis functions, etc.) some non-trival computation needs to be performed, and ideally stored for re-use every time the object is called (even in linear interpolation you may want to build a three to quickly find the closest points).

That said, I do not object to functions, the optimum is the enemy of the good :)

>  I have tried this approach in previous commits but abandoned it as I felt that since this subroutine (LoadInletProfile(args)) was changed in the newer v7.0

Protect it with a testcase... No feature is safe without a regression for it :smiling_imp:.

Reply:
> That said, I do not object to functions, the optimum is the enemy of the good :)

I agree. Now that I have the class l will on forward and make it work.

> Protect it with a testcase... No feature is safe without a regression for it 😈.

I have a regression. But after the new version release I wasn't able to merge my code because so much changed in the subroutine. Although I was level with develop most of the time.